crates/kubelet/src/kubelet.rs

```diff
@@ -187,16 +183,14 @@ async fn start_node_updater(client: kube::Client, node_name: String) -> anyhow::
 }
 
 /// Checks for shutdown signal and cleans up resources gracefully.
-async fn start_signal_handler(
-    signal: Arc<AtomicBool>,
-    client: kube::Client,
-    node_name: String,
-) -> anyhow::Result<()> {
+async fn start_signal_handler(signal: Arc<AtomicBool>) -> anyhow::Result<()> {
```

Comment:
I think this task should be removed entirely?

Reply:
I wondered about that as well but figured it was maybe in order to be able to add additional shutdown handling down the line.
Splitting this out into a handler communicating via a signal would allow implementing additional handlers that watch for a shutdown CRD, look for local files, offer a rest API or other stuff fairly easily.
Just start an additional task with the new handling that sets the shutdown AtomicBool to true.

Come to think of it, we could allow providers to define a closure that can handle custom shutdown triggers and spin up a task running this in the Kubelet. I'm not advocating for this, just a thought that came to me pretty much while writing this :)

Reply:
Just for shiggles I've written down my idea here:
https://github.com/soenkeliebau/krustlet/commit/7c8d904c4f732b2cfec72771fb7bfa0094a98e60